models/demos/t3000/mixtral8x7b/tt/mixtral_model.py

```diff
@@ -64,16 +64,21 @@ def __init__(
 
         self.compute_kernel = self.args.get_compute_kernel_config()
 
+        self.current_rot_mat, self.rot_matrix = get_single_rot_mat(self.args.head_dim, device_mesh)
+
     def forward(
         self,
         x,
         start_pos,
         current_pos,
         attn_masks,
-        rot_mats,
     ):
+        if start_pos > 0:
```

Comment:
@yieldthought Can you comment on this?

Stuti mentioned that the matmul kernel allocates a new buffer for the output, hence this makes sure that a proper deallocation is done.

Reply:
Github doesn't let me select multiple lines. The above comment is for lines 76 - 79

Reply:
@sraizada-tt can you add a comment to this section of the code mentioning this for future reference when looking at it.